x-pack/plugin/core/src/main/java/org/elasticsearch/protocol/xpack/XPackInfoRequest.java

```diff
@@ -70,6 +77,14 @@ public void setCategories(EnumSet<Category> categories) {
         return categories;
     }
 
+    public int getLicenseVersion() {
+        return licenseVersion;
+    }
+
+    public void setLicenseVersion(int licenseVersion) {
+        this.licenseVersion = licenseVersion;
+    }
```

Comment:
It is non-ideal that all the code changes in this class have to stay because of BWC. But I cannot think of a better way to avoid them.

Reply:
We don't _have to_ keep it like this. We could drop the field, ignore it on deserialization, and always write the CURRENT_VERSION on serialization, but that's probably not a net-saving.

Reply:
@ywangd I changed this to just be an ignored read + a CURRENT write. I think it ends up being neater, and it also means I could switch the version control to only do the BWC for `[7.8.1, 8.0.0)`.
Can you re-review?

Reply:
Great this change means we will be able to remove the bwc in v9.0.0 instead of carrying it over indefinitely.